v2/internal/controllers/crd_networking_privatelinkservice_test.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package controllers_test
+
+import (
+	"testing"
+
+	"github.com/Azure/go-autorest/autorest/to"
+	. "github.com/onsi/gomega"
+
+	network20201101 "github.com/Azure/azure-service-operator/v2/api/network/v1beta20201101"
+	network20220701 "github.com/Azure/azure-service-operator/v2/api/network/v1beta20220701"
+	resources "github.com/Azure/azure-service-operator/v2/api/resources/v1beta20200601"
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+)
+
+func Test_Networking_PrivateLinkService_CRUD(t *testing.T) {
+	t.Parallel()
+	tc := globalTestContext.ForTest(t)
+
+	rg := tc.CreateTestResourceGroupAndWait()
+
+	vnet := newVMVirtualNetwork(tc, testcommon.AsOwner(rg))
+
+	privateLinkServiceNetworkPolicies := network20201101.SubnetPropertiesFormat_PrivateLinkServiceNetworkPolicies_Disabled
+	subnet := newVMSubnet(tc, testcommon.AsOwner(vnet))
+	subnet.Spec.PrivateLinkServiceNetworkPolicies = &privateLinkServiceNetworkPolicies
+
+	lb, frontendIPConfigurationARMID := newLbForPLS(tc, rg, subnet)
+
+	tc.CreateResourcesAndWait(vnet, subnet, lb)
+
+	ipAllocationMethod := network20220701.IPAllocationMethod_Dynamic
+	ipVersion := network20220701.IPVersion_IPv4
+	plsConfigMap := "plsconfig"
+	pls := &network20220701.PrivateLinkService{
+		ObjectMeta: tc.MakeObjectMeta("pls"),
+		Spec: network20220701.PrivateLinkService_Spec{
+			AutoApproval: &network20220701.ResourceSet{
+				Subscriptions: []string{tc.AzureSubscription},
+			},
+			IpConfigurations: []network20220701.PrivateLinkServiceIpConfiguration{
+				{
+					Name:                      to.StringPtr("config"),
+					PrivateIPAddressVersion:   &ipVersion,
+					PrivateIPAllocationMethod: &ipAllocationMethod,
+					Subnet: &network20220701.Subnet_PrivateLinkService_SubResourceEmbedded{
+						Reference: tc.MakeReferenceFromResource(subnet),
+					},
+				},
+			},
+			LoadBalancerFrontendIpConfigurations: []network20220701.FrontendIPConfiguration_PrivateLinkService_SubResourceEmbedded{
+				{
+					Reference: &genruntime.ResourceReference{
+						ARMID: frontendIPConfigurationARMID,
+					},
+				},
+			},
+			Location: tc.AzureRegion,
+			Owner:    testcommon.AsOwner(rg),
+			Visibility: &network20220701.ResourceSet{
+				Subscriptions: []string{tc.AzureSubscription},
+			},
+			OperatorSpec: &network20220701.PrivateLinkServiceOperatorSpec{
+				ConfigMaps: &network20220701.PrivateLinkServiceOperatorConfigMaps{
+					Alias: &genruntime.ConfigMapDestination{
+						Name: plsConfigMap,
+						Key:  "alias",
```

Comment:
What exactly is alias and why do users want to consume it?

Reply:
Alias is a globally unique name for your service. It helps you mask the customer data for your service and at the same time creates an easy-to-share name for your service. When you create a Private Link service, Azure generates an alias for your service that you can share with your customers. Your customers can use this alias to request a connection to your service.

The alias is composed of three parts: Prefix.GUID.Suffix

Prefix is the service name. You can pick your own prefix. After "Alias" is created, you can't change it, so select your prefix appropriately.

GUID will be provided by platform. This GUID makes the name globally unique.

Suffix is appended by Azure: region.azure.privatelinkservice

Complete alias: Prefix. {GUID}.region.azure.privatelinkservice

https://learn.microsoft.com/en-us/azure/private-link/private-link-service-overview#alias

Reply:
This sounds like something that should be called out in our docs. Consider adding a doc fragment to `docs/v2/azure/supported-resources` so that this included.